order/src/main/java/com/yas/order/service/CheckoutService.java

```diff
@@ -33,70 +35,94 @@
+                productService.getProductInfomation(productIds, 0, productIds.size());
+        associateProductWithCheckoutItem(checkout, products, checkoutItems);
+
+        checkout.setCheckoutItems(checkoutItems);
+        return checkout;
+    }
 
-        return checkoutVm.toBuilder().checkoutItemVms(checkoutItemVms).build();
+    private void associateProductWithCheckoutItem(
+            Checkout checkout,
+            Map<Long, ProductCheckoutListVm> products,
+            List<CheckoutItem> checkoutItems) {
+        checkoutItems.forEach(t -> {
+            ProductCheckoutListVm product = products.get(t.getProductId());
+            if (product == null) {
+                throw new NotFoundException("PRODUCT_NOT_FOUND", t.getProductId());
+            } else {
+                t.setProductName(product.getName());
+                t.setProductPrice(BigDecimal.valueOf(product.getPrice()));
+                checkout.addAmount(product.getPrice() * t.getQuantity());
+            }
+        });
     }
 
     public CheckoutVm getCheckoutPendingStateWithItemsById(String id) {
 
         Checkout checkout = checkoutRepository.findByIdAndCheckoutState(id, CheckoutState.PENDING).orElseThrow(()
-            -> new NotFoundException(CHECKOUT_NOT_FOUND, id));
+                -> new NotFoundException(CHECKOUT_NOT_FOUND, id));
 
         if (!checkout.getCreatedBy().equals(AuthenticationUtils.getCurrentUserId())) {
-            throw new Forbidden(Constants.ErrorCode.FORBIDDEN);
+            throw new ForbiddenException(Constants.ErrorCode.FORBIDDEN, "You can not view this checkout");
         }
 
         CheckoutVm checkoutVm = checkoutMapper.toVm(checkout);
 
-        List<CheckoutItem> checkoutItems = checkoutItemRepository.findAllByCheckoutId(checkout.getId());
+        List<CheckoutItem> checkoutItems = checkout.getCheckoutItems();
         if (CollectionUtils.isEmpty(checkoutItems)) {
             return checkoutVm;
         }
 
         Set<CheckoutItemVm> checkoutItemVms = checkoutItems
-            .stream()
-            .map(checkoutMapper::toVm)
-            .collect(Collectors.toSet());
+                .stream()
+                .map(checkoutMapper::toVm)
+                .collect(Collectors.toSet());
 
         return checkoutVm.toBuilder().checkoutItemVms(checkoutItemVms).build();
     }
 
     public Long updateCheckoutStatus(CheckoutStatusPutVm checkoutStatusPutVm) {
         Checkout checkout = checkoutRepository.findById(checkoutStatusPutVm.checkoutId())
-            .orElseThrow(() -> new NotFoundException(CHECKOUT_NOT_FOUND, checkoutStatusPutVm.checkoutId()));
+                .orElseThrow(() -> new NotFoundException(CHECKOUT_NOT_FOUND, checkoutStatusPutVm.checkoutId()));
         checkout.setCheckoutState(CheckoutState.valueOf(checkoutStatusPutVm.checkoutStatus()));
         checkoutRepository.save(checkout);
         Order order = orderService.findOrderByCheckoutId(checkoutStatusPutVm.checkoutId());
```

Comment:
Please use logging to track key operations like checkout creation or status update, etc.